Xamarin.Essentials/Maps/Map.android.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Globalization;
+using System.Threading.Tasks;
+using Android.Content;
+using AndroidUri = Android.Net.Uri;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Map
+    {
+        internal static Task PlatformOpenMapsAsync(double latitude, double longitude, MapLaunchOptions options)
+        {
+            var uri = string.Empty;
+            if (string.IsNullOrWhiteSpace(options.Name))
+                uri = $"geo:{latitude.ToString(CultureInfo.InvariantCulture)},{longitude.ToString(CultureInfo.InvariantCulture)}?q={latitude.ToString(CultureInfo.InvariantCulture)},{longitude.ToString(CultureInfo.InvariantCulture)}";
+            else
+                uri = $"geo:{latitude.ToString(CultureInfo.InvariantCulture)},{longitude.ToString(CultureInfo.InvariantCulture)}?q={latitude.ToString(CultureInfo.InvariantCulture)},{longitude.ToString(CultureInfo.InvariantCulture)}({options.Name})";
+            StartIntent(uri);
+            return Task.CompletedTask;
+        }
+
+        internal static Task PlatformOpenMapsAsync(Placemark placemark, MapLaunchOptions options)
+        {
+            if (string.IsNullOrWhiteSpace(options.Name))
+                options.Name = string.Empty;
+
+            if (string.IsNullOrWhiteSpace(placemark.Thoroughfare))
+                placemark.Thoroughfare = string.Empty;
+
+            if (string.IsNullOrWhiteSpace(placemark.Locality))
+                placemark.Locality = string.Empty;
+
+            if (string.IsNullOrWhiteSpace(placemark.AdminArea))
+                placemark.AdminArea = string.Empty;
+
+            if (string.IsNullOrWhiteSpace(placemark.CountryName))
+                placemark.CountryName = string.Empty;
+            var uri = $"http://maps.google.com/maps?q={placemark.Thoroughfare} {placemark.Locality}, {placemark.AdminArea} {placemark.CountryName}";
```

Comment:
Should we be url encoding here?

Reply:
I added uri escaping to android and uwp because they use URIs. Please take a look at PlacemarkExtensions.shared.cs. I had to turn off a vs warning because style cop complained otherwise. The accessability modifier appears to be causing trouble.